src/typeDeclarations.ts

```diff
@@ -44,6 +52,7 @@ export interface ProviderOptions {
   debugActions?: boolean
   debugRequests?: boolean
   paramsParser?: (t: string) => string
+  parseParams?: (t: string) => string // TO REMOVE
```

Comment:
To remove? If this was intentional, discard this comment.

Reply:
Yes.. we'll remove it in the future bc it is an error that we can't remove without breaking the API.
We'll probably add deprecation warnings in the meantime...
Anyway, this PR is still a Draft, today I added a LOT more and still a draft :P

Reply:
I'll slap a WIP at the title then, to avoid further reviews while unfinished.